### ISSUE TITLE
How to use this library for exposing user errors in Modals?

### ISSUE THREAD

**Issue by stevekuznetsov** [slack-go/slack#804]
Hey! I found the [`ViewSubmissionResponse`](https://godoc.org/github.com/slack-go/slack#ViewSubmissionResponse) `struct` and the associated [`NewErrorsViewSubmissionResponse()`](https://godoc.org/github.com/slack-go/slack#NewErrorsViewSubmissionResponse) constructor but I can't quite figure out how to use this data structure to actually send the data to the Slack server and expose an error in a Modal submission as per the [Slack documentation](https://api.slack.com/surfaces/modals/using#displaying_errors). I also don't see any part of this library's codebase using this structure. Are there docs I am missing?

**Comment by stevekuznetsov** [comment]
@kanata2 @tsein-stripe I see you added the associated code in https://github.com/slack-go/slack/pull/669 - any thoughts?

**Comment by cjdenio** [comment]
Here's the snippet I use to accomplish this!

```go
w.Header().Add("Content-Type", "application/json")
  resp, _ := json.Marshal(slack.ViewSubmissionResponse{
    ResponseAction: slack.RAErrors,
    Errors: map[string]string{
      "action_id": "This field has an error :(",
    },
  })

w.Write(resp)
```

**Comment by tsein-stripe** [comment]
You can also respond with this in your view submission handler:
```go
slack.NewErrorsViewSubmissionResponse(map[string]string{
  "{block_id_to_associate_error_with}": "{error_message}",
})
```
https://github.com/slack-go/slack/blob/master/views.go#L84-L89

It's just a helper to build the struct that @cjdenio referenced above.

**Comment by stevekuznetsov** [comment]
Yep, this part of the Slack documentation is a bit hard to follow. It's not necessary for the library to explain that you send these things as a response to the webhook but it sure would save some time.  Thanks all!

**Comment by kanata2** [comment]
Thank you for your feedback :)

**Comment by mdelapenya** [comment]
Hi! 👋 It's still not clear to me how to use that object into the interaction:

```golang
resAction := slack.NewErrorsViewSubmissionResponse(map[string]string{
    "destroy-cluster-confirm": "Please type DESTROY to confirm you want to destroy your cluster",
})
```

Could you provide a more elaborate example where the Handler method uses that object? 🙏 

Thanks in advance!

**Comment by tforbes-cw** [comment]
I got confused by this process, and nobody answered the above, so here it goes:

1) Make sure you're not ack'ing the `InteractionTypeViewSubmission` prematurely.  The first ack must contain the payload
2) This will only work on InputBlocks (ref: https://api.slack.com/surfaces/modals#handling_submissions and https://github.com/slack-go/slack/issues/1013#issuecomment-1243645407)
3) The key in the errors map must match the BlockID of the target modal InputBlock

```go
func (s *SlackBot) handleInteractive(evt *socketmode.Event) {
	callback, ok := evt.Data.(slack.InteractionCallback)
	if !ok {
		log.Error().Msg("failed to cast to InteractionCallback")
		return
	}

	switch callback.Type {
	// Submit events
	case slack.InteractionTypeViewSubmission:
		s.handleInteractionSubmit(&callback, evt)
        }
}
```
Route the submission to your handler.

```go
func (s *SlackBot) ValidateSchedule(values map[string]map[string]slack.BlockAction) (errMap map[string]string, err error) {
	endDtm := time.Unix(values[EndDtmBlockID][EndDtmActionID].SelectedDateTime, 0)
	errMap = make(map[string]string)
	startDtm := time.Unix(values[StartDtmBlockID][StartDtmActionID].SelectedDateTime, 0)

	// check end date happens after start date
	if endDtm.Sub(startDtm) < 0 {
		errMap[EndDtmBlockID] = fmt.Sprintf("Validation failed: %s is before %s.\n", endDtm, startDtm)
		err = errors.New(errMap[EndDtmBlockID])

		return errMap, err
	}

	return errMap, err
}

```
Build the error map based on the block IDs

```go
func (s *SlackBot) handleInteractionSubmit(callback *slack.InteractionCallback, evt *socketmode.Event) {
	errMap, err := s.ValidateSchedule(callback.View.State.Values)
	if err != nil {
		s.slackSocket.Ack(*evt.Request, slack.NewErrorsViewSubmissionResponse(errMap))

		return
	}
```
Send the error map as a payload in the Ack
